clients/src/main/java/org/apache/kafka/common/record/FileRecords.java

```diff
@@ -297,12 +297,33 @@ public int writeTo(TransferableChannel destChannel, int offset, int length) thro
      * @param startingPosition The starting position in the file to begin searching from.
      * @return the batch's base offset, its physical position, and its size (including log overhead)
      */
-    public LogOffsetPosition searchForOffsetWithSize(long targetOffset, int startingPosition) {
+    public LogOffsetPosition searchForOffsetFromPosition(long targetOffset, int startingPosition) {
```

Comment:
Please add comments to highlight the following complex logic is to save memory by avoiding using `lastOffset`. Otherwise, there may be a developer who want to streamline the code by using `lastOffset`.